mlflow/lightgbm.py

```diff
@@ -837,22 +849,56 @@ def infer_model_signature(input_example):
                 registered_model_name=registered_model_name,
             )
 
+        # Whether to automatically log the training dataset as a dataset artifact.
+        if _log_datasets:
+            try:
+                data = train_set.data
+                label = train_set.label
+                # create a CodeDatasetSource
+                context_tags = context_registry.resolve_tags()
+                source = CodeDatasetSource(
+                    mlflow_source_type=context_tags[MLFLOW_SOURCE_TYPE],
+                    mlflow_source_name=context_tags[MLFLOW_SOURCE_NAME],
+                )
+
+                # create a dataset
+                if isinstance(data, pd.DataFrame):
+                    dataset = from_pandas(df=data, source=source)
+                else:
```

Comment:
what is the expected type of `data` in this case?

Ideally, it would be nice to make this an `elif isinstance(data, <something else>)` check. That way, we can add a catch-all else clause that aborts dataset autologging and prints a warning: "unrecognized dataset type, dataset information won't be logged for this model training session"